undertow/src/main/java/org/wildfly/extension/undertow/ByteBufferPoolDefinition.java

```diff
@@ -0,0 +1,184 @@
+            .build();
+
+    protected static final SimpleAttributeDefinition DIRECT = new SimpleAttributeDefinitionBuilder(Constants.DIRECT, ModelType.BOOLEAN)
+            .setRequired(false)
+            .setRestartAllServices()
+            .setAllowExpression(true)
+            .build();
+
+    protected static final SimpleAttributeDefinition THREAD_LOCAL_CACHE_SIZE = new SimpleAttributeDefinitionBuilder(Constants.THREAD_LOCAL_CACHE_SIZE, ModelType.INT)
+            .setRequired(false)
+            .setRestartAllServices()
+            .setValidator(new IntRangeValidator(0, true, true))
+            .setAllowExpression(true)
+            .setDefaultValue(new ModelNode(12))
+            .build();
+
+
+    protected static final SimpleAttributeDefinition LEAK_DETECTION_PERCENT = new SimpleAttributeDefinitionBuilder(Constants.LEAK_DETECTION_PERCENT, ModelType.INT)
+            .setRequired(false)
+            .setRestartAllServices()
+            .setValidator(new IntRangeValidator(0, true, true))
+            .setAllowExpression(true)
+            .setDefaultValue(new ModelNode(0))
+            .build();
+
+    private static final List<AttributeDefinition> ATTRIBUTES = Arrays.asList(BUFFER_SIZE, MAX_POOL_SIZE, DIRECT, THREAD_LOCAL_CACHE_SIZE, LEAK_DETECTION_PERCENT);
+
+    private ByteBufferPoolDefinition() {
+        super(UndertowExtension.BYTE_BUFFER_POOL_PATH,
+                UndertowExtension.getResolver(Constants.BYTE_BUFFER_POOL),
+                new BufferPoolAdd(),
+                new ReloadRequiredRemoveStepHandler()
+        );
+    }
+
+    @Override
+    public void registerCapabilities(ManagementResourceRegistration resourceRegistration) {
+        resourceRegistration.registerCapability(UNDERTOW_BUFFER_POOL_RUNTIME_CAPABILITY);
+    }
+
+    @Override
+    public Collection<AttributeDefinition> getAttributes() {
+        return ATTRIBUTES;
+    }
+
+
+    private static class BufferPoolAdd extends AbstractAddStepHandler {
+
+        private BufferPoolAdd() {
+            super(ByteBufferPoolDefinition.ATTRIBUTES);
+        }
+
+        @Override
+        protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+            final PathAddress address = PathAddress.pathAddress(operation.get(OP_ADDR));
+            final ModelNode bufferSizeModel = BUFFER_SIZE.resolveModelAttribute(context, model);
+            final ModelNode maxPoolSizeModel = MAX_POOL_SIZE.resolveModelAttribute(context, model);
+            final ModelNode directModel = DIRECT.resolveModelAttribute(context, model);
+            final int threadLocalCacheSize = THREAD_LOCAL_CACHE_SIZE.resolveModelAttribute(context, model).asInt();
+            final int leakDetectionPercent = LEAK_DETECTION_PERCENT.resolveModelAttribute(context, model).asInt();
+
+            final int bufferSize = bufferSizeModel.isDefined() ? bufferSizeModel.asInt() : defaultBufferSize;
```

Comment:
no need to change this but just fyi this and the others can be

final int bufferSize = bufferSizeModel.asInt(defaultBufferSize);